mashup.js

```diff
@@ -0,0 +1,144 @@
+
+Article.prototype.byLine = function() {
+    var byLine = this.article.fields.byline;
+    if ( byLine === undefined ) {
+        byLine = 'theguardian.com';
+    }
+    return byLine;
+};
+
+Article.prototype.articleDate = function() {
+    var articleDate = new ArticleDate(this.article.webPublicationDate);
+    return articleDate.toArticleDateString();
+};
+
+Article.prototype.summary = function() {
+    var summary = this.article.blocks.body[0].bodyTextSummary.substring(0,415);
+    // assumes there is a blank
+    return summary.substring(0,summary.lastIndexOf(' '));
+};
+
+Article.prototype.genHtml = function() {
+    var htmlString = '<article>';
+    htmlString += '<h2>' + this.article.webTitle + '</h2>';
+    htmlString += '<h3>' + this.byLine() + '</h3><div class="separator"></div>';
+    htmlString += '<h3>' + this.articleDate() + '</h3>';
+    htmlString += '<div class="clear"><img src="' + this.article.fields.thumbnail + '"/>';
+    htmlString += '<p>' + this.summary() + ' ...';
+    htmlString += '<a href="' + this.article.webUrl + '">Read the full article</a></p></div>';
+    htmlString += '</article>';
+    return htmlString;
+};
+
+var displayNews = function(event,data) {
+    var count = 0;
+    for (var i = 0; i < data.response.results.length && count < 10; i++ ) {
+        switch (data.response.results[i].type) {
+            case 'article':
+                var article = new Article(data.response.results[i]);
+                if ( article.valid() ) {
+                    count++;
+                    $(event.data.newsSelectorId).append(article.genHtml());
+                }
+                break;
+            // add cases for other types
+        }
+    }
+    $(event.data.newsSelectorId).css('display','block');
+    $(event.data.closeSelectorId).css('visibility','visible');
+};
+
+var displayErrMsg = function() {
+    $('#msg').remove();
+    $('body').append('<p id="msg">*** Issue with response ***</p>');
+};
+
+var openHandler = function(event) {
+    if ( $(event.data.newsSelectorId).css('display') === 'none' ) {
+        $.ajax({
+            url: event.data.url,
+            success: function(data) {
+                if (data.response.status === "ok") {
+                    displayNews(event,data);
+                } else {
+                    displayErrMsg();
+                }
+            }
+        });
+    }
+};
+
+var closeHandler = function(event) {
+    $(this).css('visibility','hidden');
+    $(event.data.newsSelectorId).slideUp('slow').empty();
+};
+
+var setUp = function() {
+    var categories = [
+        'Mammal','Bird','Reptile','Fish','Amphibian','Insect','Worm','Crustacean','Coral','Protozoan','Dinosaur'
+    ];
+    for (var i = 0; i < categories.length; i++) {
+        var category = new Category(categories[i]);
+        $('main').append(category.genHtml());
+        $(category.selectorId).click(category,openHandler);
+        $(category.closeSelectorId).click(category,closeHandler);
+    }
+};
```

Comment:
I would probably put all of these methods on a class/object as well, e.g. `App`, so that they are a) grouped and b) not globals.